app/src/main/java/com/portfolio/tasky/networking/usecases/TaskyCallStatusImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.portfolio.tasky.networking.usecases
+
+import android.content.Context
+import android.widget.Toast
+import com.portfolio.tasky.networking.usecases.domain.TaskyLoader
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import javax.inject.Inject
+
+/**
+ * TaskStatusImpl will harbour the code for un-successful api calls and
+ * would present errors, messages in dialogs
+ * with respect to Response codes.
+ */
+class TaskyCallStatusImpl @Inject constructor(private val taskyLoader: TaskyLoader, private val context: Context) : TaskyCallStatus {
```

Comment:
This whole class won't be needed with coroutines

Reply:
This is required since I want to handle error for api in dialogs over here.

Reply:
Won't be needed when using Coroutines, since you can basically return a true/false (this is simplified) to the viewmodel, and then the viewmodel can show an error or success

And in your viewmodel, before calling the repository for login/register, you can set the isLoading to true, and on success or failure from repository, you can set the isLoading to false

So you do state managing inside the viewmodel, and api call inside the repository

Remember, the jobs are:
UI = Display the state, and modify the UI based on the state. Send user interactions to the viewmodel (Ex: User clicked a button)
ViewModel = Modify the state based on the user interactions or something like an api/db call from repository
Repository = Make api and db calls by accessing the retrofit api or room dao. This information will then be returned to the viewmodel

Reply:
This explanation clears the picture 👍